### PR TITLE
Added docs for creating middleware

### PR DESCRIPTION
**Pull request by NicoPlyley**
Hi @yusukebe,

I added what was mentioned in #317 to the existing custom middleware section in the guide. I tried to keep it short but informative. I was debating adding another example for adding parameters to custom middleware, but I figured linking to the `createMiddleware` factory would give a user the answer they are looking for.

Let me know if you want any updates or changes!

closes: #317

### CODE REVIEW

File: guides/middleware.md [hunk 1]
```diff
@@ -133,6 +133,39 @@ app.use('/message/*', async (c, next) => {
...
+```ts
+import { createMiddleware } from 'hono/factory'
+
+const logger = createMiddleware(async (c, next) => {
+  console.log(`[${c.req.method}] ${c.req.url}`)
+  await next()
+})
+
+```
+Additionally, middleware can be designed to modify responses if necessary:
```
Comment (yusukebe) [inline guides/middleware.md:151]: I think it's better to separate this description from here and make it a sub-section of "Custom Middleware," which is put in.  The title will be like "Modify the Response after the `next()`".

Comment (NicoPlyley) [reply guides/middleware.md:151]: Sounds good, I will update that soon

Comment (NicoPlyley) [reply guides/middleware.md:159]: Fixed: 3abe563

Comment (yusukebe) [reply guides/middleware.md:159]: Cool!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@NicoPlyley 

Thanks! I'll merge now.

**Pull request merged**